CoEpi/ble/Peripheral.swift

```diff
@@ -1,33 +1,43 @@
-import Foundation
 import CoreBluetooth
+import Dip
+import Foundation
 import os.log
 import RxRelay
 
+struct Contact2 {
```

Comment:
?

Reply:
The point is to be able to pass the CEN data to consumers of the BT code, but not have the BT code know that it's CEN data/know what it is. So the BT code can just pass it out as Data without knowing what data it is. That's what this comment was about:
next step would be to make Peripheral generic on theirData and ourData

So the consumer can know that it's actually CENs but the Peripheral code just sees it as data

Reply:
At the moment all what we get from BT (correct me if I'm wrong) is a byte array, which we did read from their characteristic or advertisement data.

Either return only the byte array and convert to a CEN type (this would be just a wrapper for a byte array/data) or couple BT directly with CEN.
I don't find the later so bad, given that the covadwitch plans to do the same and it helps us a little with readability and safety. The way the BT is used is tailored around contact tracing, so this coupling makes sense.

Reply:
I think this function is the delegate called from BT - `func peripheralManager(_ peripheral: CBPeripheralManager, didReceiveRead request: CBATTRequest)` I used `Data` here because that's the type of `request.value`. I think on the outside we can convert the data to a CEN type

Reply:
For removing all the `Contact` classes, do you mean all of these as well?
https://github.com/Co-Epi/app-ios/blob/develop/CoEpi/domain/model/Contact.swift
https://github.com/Co-Epi/app-ios/blob/develop/CoEpi/repo/realm/model/RealmContact.swift
https://github.com/Co-Epi/app-ios/blob/develop/CoEpi/repo/ContactRepo.swift
https://github.com/Co-Epi/app-ios/blob/develop/CoEpi/repo/realm/RealmContactRepo.swift

Reply:
Seems that for read requests, request.value is null:
`
/**
         2020-03-31 17:38:35.724037+0200 CoEpi[2303:2652797] [BLEPeripheral] Peripheral manager did receive read request:
         <CBATTRequest: 0x2803d81c0
             Central = <CBCentral: 0x2803d8480
                 identifier = 9E91D45A-64F4-1E48-E459-5C9068A36BBD,
                 MTU = 182
             >,
             Characteristic = <CBMutableCharacteristic: 0x2830b38e0
                 UUID = 2AC35B0B-00B5-4AF2-A50E-8412BCB94285,
                 Value = (null),
                 Properties = 0x2,
                 Permissions = 0x1,
                 Descriptors = (null),
                 SubscribedCentrals = (
                 )
             >,
             Offset = 0, Value = (null)
         >*/
`

Reply:
@johnsonh Yeah. Let's update with the API models (and the Android app, which orients with those). The `Contact` classes are mostly experimental.

- `Cen`
- `CenKey`
- `CenReport` (we don't need this for 0.1)

Additionally on Android I refined the `Cen` data type a bit. It seems that the timestamp is used only for _observed_ CENs. So I renamed the CEN+timestamp datatype in `ReceivedCen`. The CEN in `ReceivedCen` would be the actual `Cen` data type.

```kotlin
data class Cen(val bytes: ByteArray)
data class ReceivedCen(val cen: Cen, val timestamp: Int)
```

Reply:
Ok, I've removed `ContactOld`. I'd propose that we remove `Contact` related code in a separate PR, since this one is already huge and that seems like a discrete set of work.

Reply:
So you're saying we should
- Update `Cen` to be `ReceivedCen`, like in Android
- Add API models?

I can do the former, but can we do the latter in a follow up PR?

Reply:
Added a TODO section in the PR description

Reply:
I'll approve. You can decide when you want to do these changes.